plugins/modules/pacman_key.py

```diff
@@ -129,12 +135,55 @@
 from ansible.module_utils.common.text.converters import to_native
 
 
+class GpgListResult(object):
+    """Wraps gpg --list-* output."""
+
+    def __init__(self, line):
+        self._parts = line.split(':')
+
+    @property
+    def kind(self):
+        return self._parts[0]
+
+    @property
+    def valid(self):
+        return self._parts[1]
+
+    @property
+    def is_fully_valid(self):
+        return self.valid == 'f'
+
+    @property
+    def key(self):
+        return self._parts[4]
+
+    @property
+    def user_id(self):
+        return self._parts[9]
+
+
+def gpg_get_first(lines, kind, attr):
+    for line in lines:
+        glr = GpgListResult(line)
+        if glr.kind == kind:
+            return getattr(glr, attr)
```

Comment:
And that would make this one:
```suggestion
def gpg_get_first(lines, kind, attr):
    for glr in gpg_gather_all(lines):
        if glr.kind == kind:
            return getattr(glr, attr)
```